crates/sqllib/src/casts.rs

```diff
@@ -1069,103 +1043,126 @@ pub fn cast_to_s_Time(value: Time, size: i32, fixed: bool) -> String {
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_s_i16(value: i16, size: i32, fixed: bool) -> String {
+pub fn cast_to_s_i16(value: i16, size: i32, fixed: bool) -> SqlResult<String> {
     let result = value.to_string();
     limit_or_size_string(result, size, fixed)
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_s_i32(value: i32, size: i32, fixed: bool) -> String {
+pub fn cast_to_s_i32(value: i32, size: i32, fixed: bool) -> SqlResult<String> {
     let result = value.to_string();
     limit_or_size_string(result, size, fixed)
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_s_i64(value: i64, size: i32, fixed: bool) -> String {
+pub fn cast_to_s_i64(value: i64, size: i32, fixed: bool) -> SqlResult<String> {
     let result = value.to_string();
     limit_or_size_string(result, size, fixed)
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_s_u(value: usize, size: i32, fixed: bool) -> String {
+pub fn cast_to_s_u(value: usize, size: i32, fixed: bool) -> SqlResult<String> {
     let result = value.to_string();
     limit_or_size_string(result, size, fixed)
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_s_V(value: Variant, size: i32, fixed: bool) -> String {
+pub fn cast_to_s_V(value: Variant, size: i32, fixed: bool) -> SqlResult<String> {
+    // This function should never be called
     let result: String = value.try_into().unwrap();
     limit_or_size_string(result, size, fixed)
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_s_VN(value: Option<Variant>, size: i32, fixed: bool) -> Option<String> {
-    Some(cast_to_s_V(value.unwrap(), size, fixed))
+pub fn cast_to_s_VN(value: Option<Variant>, size: i32, fixed: bool) -> SqlResult<String> {
+    // This function should never be called
+    cast_to_s_V(value.unwrap(), size, fixed)
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_sN_V(value: Variant, size: i32, fixed: bool) -> Option<String> {
-    let result: String = value.try_into().ok()?;
-    Some(limit_or_size_string(result, size, fixed))
+pub fn cast_to_sN_V(value: Variant, size: i32, fixed: bool) -> SqlResult<Option<String>> {
+    let result: Result<String, _> = value.try_into();
+    match result {
+        Err(_) => Ok(None),
+        Ok(result) => r2o(limit_or_size_string(result, size, fixed)),
+    }
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_sN_VN(value: Option<Variant>, size: i32, fixed: bool) -> Option<String> {
-    let value = value?;
-    cast_to_sN_V(value, size, fixed)
+pub fn cast_to_sN_VN(value: Option<Variant>, size: i32, fixed: bool) -> SqlResult<Option<String>> {
+    match value {
+        None => Ok(None),
+        Some(value) => cast_to_sN_V(value, size, fixed),
+    }
 }
 
 #[doc(hidden)]
 #[inline]
-pub fn cast_to_bytes_V(value: Variant, size: i32) -> ByteArray {
-    let result: ByteArray = value.try_into().unwrap();
-    ByteArray::with_size(result.as_slice(), size)
+pub fn cast_to_bytes_V(value: Variant, size: i32) -> SqlResult<ByteArray> {
+    // Should never be called
```

Comment:
why not?

Reply:
maybe it should be unreachable!() then?

Reply:
In general a cast from variant should always produce a nullable result.
So only the function which returns Option should be called.
But Calcite is full of surprises.